addons/crm/models/crm_lead.py

```diff
@@ -832,7 +834,17 @@ def action_set_won(self):
         # group the leads by team_id, in order to write once by values couple (each write leads to frequency increment)
         leads_by_won_stage = {}
         for lead in self:
-            stage_id = lead._stage_find(domain=[('is_won', '=', True)])
+            won_stages = self._stage_find(domain=[('is_won', '=', True)], limit=None)
+            # ABD : In Fine, we would like to prevent a "not won" stage to have a higher
+            # sequence than any "is won" stage. But while this is not the case, searching
+            # for the "is won" stage while alterning the sequence order will correctly
+            # handle the following case :
+            #       stage sequence : [x] [x is won] [y] [y is won] [z] [z is won]
+            #       when in [y] and marked as "won", should go to [y is won]
+            #       and not in [x is won] nor [z is won]
```

Comment:
I don't understand this note, we're only looking for `is_won` leads, where would the "not won" matches come from?

Or is this an explanation for the fallback, where we seek backwards if there is no match forwards?

Reply:
This is an explanation why we seek forward from the current stage to stages with higher sequence, then if there is no match, backward to stages with lower sequence. Also, I left this note because in the future if we decide to prevent alternating [not won] stages with [won] stages, then we can revert back to the previous code (less complex)

Reply:
It seems we could add a line telling we could have a mixed pipeline, with "won" stage being separated by "standard" stages (even if not ideal, it may happen). This would help understanding the remaining of the comment.